src/components/membershipCard/MembershipCardTermsModal.tsx

```diff
@@ -77,7 +78,7 @@ const MembershipCardTermsModal: React.FC<MembershipCardTermsModalProps> = ({
     }
   }
 
-  const generateProductLink = (details: { productName: string; id: string }) => {
+  const generateProductLink = (details: { productName: string; id: string; creatorId: string }) => {
```

Comment:
since id and creatorId are just variations of the same thing (a unique identifier) ,maybe we can combine them as `resourceId` to keep it simple